xinference/core/supervisor.py

```diff
@@ -243,25 +255,37 @@ def get_model_registration(
 
     @log_async(logger=logger)
     async def register_model(self, model_type: str, model: str, persist: bool):
-        if model_type == "LLM":
+        if model_type == "LLM" or model_type == "embedding":
```

Comment:
Better use different code branches to handle different model types, otherwise it would be complex when types become more and more.